api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -144,4 +145,33 @@
      * @return the configuration sources
      */
     Iterable<ConfigSource> getConfigSources();
+
+    /**
+     * Convert an abritrary value to the specified type using the converters known to this instance
+     * 
+     * @param <T>
+     *             the conversion type
+     * @param value
+     *             the value to convert
+     * @param type
+     *             the type into which the value should be converted
+     * @return the converted value
+     * @throws java.lang.IllegalArgumentException if the value cannot be converted to the specified type.
+     */
+    default <T> T convert(String value, Class<T> type) {        
+        return value == null ? null
+                             : getConverter(type).orElseThrow(() -> new IllegalArgumentException("no converter available for " + type))
+                                                 .convert(value);
+    }
+
+    /**
+     * Return the {@link Converter} used by this instance to produce instances of the specified type from string values. 
+     * 
+     * @param <T>
+     *             the conversion type
+     * @param forType
+     *             the type to be produced by the converter
+     * @return an {@link Optional} containing the converter, or empty if no converter is available for the specified type
+     */
+    <T> Optional<Converter<T>> getConverter(Class<T> forType);
```

Comment:
Shouldn't we also expose a `<T> Converter<T> getConverter`?

Reply:
This was discussed previously on a call I think, and the consensus was that the optional variant is more consistent with the other API elements.  It allows the caller to decide whether to throw an exception, return null, or stick with the `Optional` if there is no converter available, all with one method.